File: 80_Remove_Duplicates_from_Sorted_Array_II.c
# include <stdio.h>
# include <stdlib.h>

int removeDuplicates(int* nums, int numsSize) {
    int uniqueCount = 1, count = 1;
    if (numsSize == 0) return 0;
    
    for (int i = 1; i < numsSize; i++) {
        if (nums[i] == nums[i-1]){
            count++;
        }
        else {
            count = 1;
        }

        if (count > 2) {
            continue;
        }
        uniqueCount++;
        nums[uniqueCount - 1] = nums[i];
    }

    return uniqueCount;
}

int main() {
    int nums[] = {1,1,1,2,2,3};
    int size = sizeof(nums) / sizeof(nums[0]);
    
    int newSize = removeDuplicates(nums, size);
    
    printf("New size: %d\n", newSize);
    printf("Array after removing duplicates: ");
    for (int i = 0; i < newSize; i++) {
        printf("%d ", nums[i]);
    }
    printf("\n");
    
    return 0;
}

File: 80_Remove_Duplicates_from_Sorted_Array_II.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        if not nums:
            return 0

        write_index = 1
        count = 1

        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                count += 1
            else:
                count = 1

            if count <= 2:
                nums[write_index] = nums[i]
                write_index += 1

        return write_index
    
# Example usage:
a = Solution()
print(a.removeDuplicates([1,1,1,2,2,3]))  # Output: 5
print(a.removeDuplicates([0,0,1,1,1,1,2,3,3]))  # Output: 7
        